command/alloc_status.go

```diff
@@ -115,7 +115,7 @@ func (c *AllocStatusCommand) Run(args []string) int {
 	}
 
 	// If args not specified but output format is specified, format and output the allocations data list
-	if len(args) == 0 && json || len(tmpl) > 0 {
+	if len(args) == 0 && (json || len(tmpl) > 0) {
```

Comment:
we only want to do this if there were no args... previous parsing would go into list mode with args if template was specified, which i think was not the intended behavior. same below for `eval status` and `job inspect`.